src/lib/interp.rs

```diff
@@ -64,8 +65,21 @@ impl VM {
         self.enter_main();
         let mut result = None;
         loop {
-            let instr = self.bytecode.bytecode[self.pc].clone();
-            match instr {
+            let bytecode_size = self.bytecode.bytecode.len();
+            if self.pc >= bytecode_size {
```

Comment:
I'm not quite sure I know what this `if` is doing. The `match` doesn't seem to be doing anything?

Reply:
This is a (somewhat hacky) way of checking if an exception has been raised and never been caught along the way.

The frame stack should always contain at least the `main` frame unless an exception is raised which is never caught, at which point all the frames are popped.